### PR TITLE
Gentec-eo Blu support and test suite

### PR DESCRIPTION
**Pull request by trappitsch**
To mix it up... here's a new instrument instead of a test

Added support for gentec-eo's Blu series power meters. These power
meters connect via bluetooth, the bluetooth dongle showsvup as a regular
serial port. In addition, they can be plugged into a USB port and vshow up
as a regular serial interface as well.

Full test suite added as well.

All routines have example doc strings with detailed examples.

Added to docs.

### CODE REVIEW

**Comment by scasagrande** [comment]
You're a champion

I'm going to have to leave this until tomorrow, it's well past my bedtime so I won't be able to effectively review this tonight :sleeping:

**Comment by trappitsch** [comment]
Same here, this was the last one for tonight... Thanks for all your work reviewing these PRs!

**Comment by trappitsch** [comment]
@scasagrande Did you have a chance to look at this? No worries if not / you're busy otherwise at the moment. I finally got some time again to think about these things, so just figured I'll check in

**Comment by scasagrande** [comment]
Oh dang it, I totally forgot. I'll look at this today.

File: instruments/gentec_eo/blu.py [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+        _terminator = self.terminator
+        self.terminator = ""
+        _timeout = self.timeout
+        self.timeout = u.Quantity(1, u.s)
+
+        # get the response
+        resp = self._no_ack_query("*DVS").split('\r\n')
+
+        # set back terminator and 3 second timeout
+        self.terminator = _terminator
```
Comment (scasagrande) [inline instruments/gentec_eo/blu.py:192]: Potential issue here: if the above line (189) errors out, then our temporary variable changes here are never reset

Comment (trappitsch) [reply instruments/gentec_eo/blu.py]: Great catch! Taking care of this now with a try, finally block to reset the temporary variables even if the read errors out. Also added the appropriate test to ensure this is actually working.

File: instruments/gentec_eo/blu.py [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+        """
+        self.sendcmd("*SSU")
+
+    # PRIVATE METHODS #
+
+    def _no_ack_query(self, cmd, size=-1):
+        """Query a value and don't expect an ACK message."""
+        self._ack_message = None
+        value = self.query(cmd, size=size)
+        self._ack_message = "ACK"
```
[the remark this answers is not in the document]
Comment (trappitsch) [reply instruments/gentec_eo/blu.py]: Exactly same solution as above, try, finally block plus test.

**Comment by scasagrande** [comment]
lgtm now! I might have to poke travis since I'm not sure why its struggling to report back to github

**Comment by trappitsch** [comment]
Yeah, saw that with Travis... it was hanging recently as well, no? Anyway, thanks for reviewing! More tests to come soon :smiley:

**Comment by scasagrande** [comment]
Well I can see on travis that it did pass, its just not reporting back correctly, so I'll just bypass this :P

**Pull request merged**